packages/devtools_app/lib/src/screens/memory/memory_protocol.dart

```diff
@@ -23,51 +23,51 @@ import 'memory_timeline.dart';
 class MemoryTracker {
   MemoryTracker(this.serviceManager, this.memoryController);
 
-  ServiceConnectionManager serviceManager;
+  ServiceConnectionManager? serviceManager;
 
   final MemoryController memoryController;
 
-  VmServiceWrapper get service => serviceManager?.service;
+  VmServiceWrapper? get service => serviceManager?.service;
 
-  Timer _pollingTimer;
+  Timer? _pollingTimer;
 
-  final Map<String, MemoryUsage> isolateHeaps = <String, MemoryUsage>{};
+  final Map<String, MemoryUsage> isolateHeaps = {};
 
   /// Polled VM current RSS.
-  int processRss;
+  int? processRss;
 
   /// Polled adb dumpsys meminfo values.
-  AdbMemoryInfo adbMemoryInfo;
+  AdbMemoryInfo? adbMemoryInfo;
 
   /// Polled engine's RasterCache estimates.
-  RasterCache rasterCache;
+  RasterCache? rasterCache;
 
   bool get hasConnection => service != null;
```

Comment:
remove this. Use serviceManager.hasConnection instead

Reply:
I did not get this. rasterCache is used to create HeapSample:

```
final HeapSample sample = HeapSample(...
```

How can I use serviceManager.hasConnection for this?

Reply:
done